### PR TITLE
Bugfix: Error from unpublished Token ATM Log on Students Page

### LINKED ISSUE #88: Check that Token ATM Log Canvas Assignment is published
The Token ATM Log Canvas Assignment must be published for student request records to be read, written, or accessed.

Token ATM should check that and warn if this necessary Canvas Assignment is/isn't published.

### PR DESCRIPTION
**Pull request by epsilonError**
### Description

This pull request fixes an error where an unpublished Token ATM Log prevents the Students page from loading Student Information and causes a global error to be shown to the user.

This is fixed by asking the user if they want Token ATM to publish the Log assignment for them. And if they don't want it published, showing 'Click to View' as the entries in the Token Balance column of the student list.

Only the API call that collects student grades for the overview list is effected by having an unpublished assignment. The detailed view for each student has no trouble with unpublished assignments.

Some grammar, formatting, and typo changes were also made as part of this bugfix.

### Testing Note

Please test if the bugfix above works as expected. Specifically,
- Confirm the Publish confirmation modal appears when opening the Students Page and the Token ATM Log isn't published.
- Confirm the detailed student record view is accessible and continues to work with or without Token ATM Log being published.

### CODE REVIEW

File: src/app/components/student-list/student-list.component.ts [hunk 1]
```diff
@@ -67,10 +70,17 @@ export class StudentListComponent implements CourseConfigurable {
             await this.refreshStudentList(this.studentSearchTerm);
         }
     }
-    //Configure the course information
+
+    // Configure the course information
     async configureCourse(course: Course): Promise<void> {
         this.course = course;
         this.configuration = await this.tokenATMConfigurationManagerService.getTokenATMConfiguration(this.course);
+        this.isLogPublished = await checkAndConfirmTokenATMLogPublished(
```
Comment (yongkanm) [inline src/app/components/student-list/student-list.component.ts:78]: Unlike request processing, which checks the publish state of Token ATM upon action, the Students page caches the publish state here. It means that if the user choose to publish the Token ATM Log by themselves, the Students page won't refresh the publish state until next time the user visit it. It is recommended to add a line of warning at the top of the page to let the user know that they need to refresh the page (by navigating to another page and navigating back).

**Review by yongkanm — CHANGES_REQUESTED** [review]
- [x] Confirm the Publish confirmation modal appears when opening the Students Page and the Token ATM Log isn't published.

    The confirmation modal displays as expected in case that Token ATM Log is not published. Both action buttons of the modal works as expected.
- [x] Confirm the detailed student record view is accessible and continues to work with or without Token ATM Log being published.
    
    <del>Testing fails. The detailed student record view could still be displayed as expected. However, if Token ATM Log is not published, neither manual balance adjustment nor making request on the student's behalf will work due to the 403 error caused by line `120` of `student-record-manager.service.ts` (not allowed to post comment to a submission of an unpublished assignment). It is recommended to be solved by passing the publish state of Token ATM Log to the `StudentRecordDisplayComponent` component, which could then be used to disable buttons that will cause error. It would be better to add a tooltip to disabled buttons or a line of warning at the top to inform the user.</del> Fixed.

**Comment by epsilonError** [comment]
> If Token ATM Log is not published, neither manual balance adjustment nor making request on the student's behalf will work … . It is recommended to be solved by passing the publish state of Token ATM Log to the `StudentRecordDisplayComponent` component, which could then be used to disable buttons that will cause error. 

Change made in recent commits. The components default to believing the Token ATM Log is not published, but there shouldn't be a flash of disabled buttons since the `collecting info from canvas...` toast appears first.


> It would be better to add a tooltip to disabled buttons or a line of warning at the top to inform the user.

Tooltips and an information paragraph have been added to the top. Regarding tooltips, something seems to prevent buttons from using the `tooltip` attribute correctly. I couldn't get them to appear, instead I applied the tooltip to a containing div (and added a div if one didn't exist).

**Review by yongkanm — APPROVED** [review]
The discovered issue is fixed. This bugfix works as expected.

**Comment by yongkanm** [comment]
> Regarding tooltips, something seems to prevent buttons from using the tooltip attribute correctly.

This is the expected behavior since disabled button does not fire `hover` events (see valor-software/ngx-bootstrap#5074).

**Pull request merged**